### PR TITLE
mail-filter/spamassassin: Bump to v3.4.3

### PR DESCRIPTION
**Pull request by Philippe23**
mail-filter/spamassassin: Bump to v3.4.3

 - Remove 3.4.2 patches that have been fixed by 3.4.3
 - Adjust SQL Update warning trigger as 3.4.3 has more schema changes

Bug: https://bugs.gentoo.org/702594
Closes: https://github.com/gentoo/gentoo/pull/13955
Package-Manager: Portage-2.3.79, Repoman-2.3.16
Signed-off-by: Philippe Chaintreuil <gentoo_bugs_peep@parallaxshift.com>

### CODE REVIEW

**Comment by Philippe23** [comment]
Just found out that the SQL schema changed again from 3.4.2 to 3.4.3 so I updated the SQL warnings version checks.

**Comment by Philippe23** [comment]
I'm going to address the GLEP-81 QA-check separately, since I'd like the CVE fix to be independent of any GLEP-81 changes.

File: mail-filter/spamassassin/spamassassin-3.4.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,284 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (juippis) [inline mail-filter/spamassassin/spamassassin-3.4.3.ebuild]: 7 should work here, could you update and test whether it does?

File: mail-filter/spamassassin/spamassassin-3.4.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+inherit perl-functions systemd toolchain-funcs user eapi7-ver
```
Comment (juippis) [inline mail-filter/spamassassin/spamassassin-3.4.3.ebuild]: ... and if EAPI-7 works, you can remove `eapi7-ver` here. I didn't see it being used anyway in this ebuild.

**Comment by orlitzky** [comment]
I pulled this one as-is. Philippe was already working on EAPI=7 and GLEP81 in another pull request before this release was made to fix the CVE. Even switching to EAPI=7 is not straightforward, because all of the build dependencies need to be overhauled to take BDEPEND into account. Since this will be sent straight to stable (it contains a security fix) we'd better not experiment too much with it.

**Comment by juippis** [comment]
Thanks @orlitzky 

wasn't aware of any history, @Philippe23 asked to take a look in IRC for fast merge so I did ;)

**Comment by Philippe23** [comment]
Yeah, my bad, I should have linked to the old PR for a broader picture.

**Pull request closed without merging**